iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothubservices/errorinjection/SendMessagesErrInjTests.java

```diff
@@ -281,6 +281,68 @@ public void sendMessagesWithTcpConnectionDropNotifiesUserIfRetryExpires() throws
         testInstance.client.setRetryPolicy(new ExponentialBackoffWithJitter());
     }
 
+    @Test
+    public void clientCanReopenAfterDisconnectionEvent() throws Exception
+    {
+        if (testInstance.protocol == HTTPS || testInstance.useHttpProxy)
+        {
+            return;
+        }
+
+        this.testInstance.setup();
+
+        this.testInstance.client.setRetryPolicy(new NoRetry());
```

Comment:
We would also need tests where the device client has a set retry policy; and the application reconnects and recovers on retry policy expiry.

Reply:
I'll add that, and a similar test for when the device operation timeout causes the disconnection

Reply:
Actually, this kind of test isn't possible to write right now. The fault injection framework we use does not allow us to cause a retryable error for X number of seconds. That functionality is necessary for us to test running through the sdk level retry and some application level retry. We only get DISCONNECTED_RETRYING followed by a successful reconnect within the SDK retry logic